lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,22 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
I'm +1 on this approach. You asked for it to be enabled, puppet makes it so regardless of what state it was in previously.